src/lib.rs

```diff
@@ -33,7 +34,7 @@ pub use query::{
 
 /// Test a release for semver violations.
 #[non_exhaustive]
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, Serialize)]
```

Comment:
One drawback to bin tests is that we need `Serialize` even in a non `#[cfg(test)]` environment.  I don't know how big a deal that is, though.

Reply:
Is that because `Serialize` is used by `cargo-insta` outside of `cfg(test)` somehow? Or is something else going on that I couldn't spot?

Reply:
Since we're running the tests in the binary part of the crate, the library side is not compiled with `#[cfg(test)]` (when it's used as a dependency for this part), so it needs to be unconditionally `Serialize`.